flink-autoscaler/src/main/java/org/apache/flink/autoscaler/config/AutoScalerOptions.java

```diff
@@ -250,6 +252,39 @@ private static ConfigOptions.OptionBuilder autoScalerConfig(String key) {
                     .withDescription(
                             "Max allowed percentage of heap usage during scaling operations. Autoscaling will be paused if the heap usage exceeds this threshold.");
 
+    public static final ConfigOption<Boolean> MEMORY_TUNING_ENABLED =
+            autoScalerConfig("memory.tuning.enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withFallbackKeys(oldOperatorConfigKey("memory.tuning.enabled"))
+                    .withDescription(
+                            "If enabled, the initial amount of memory specified for TaskManagers will be reduced according to the observed needs.");
+
+    public static final ConfigOption<MemoryTuning.HeapTuningTarget> MEMORY_TUNING_HEAP_TARGET =
+            autoScalerConfig("memory.tuning.heap.target-usage")
+                    .enumType(MemoryTuning.HeapTuningTarget.class)
+                    .defaultValue(MemoryTuning.HeapTuningTarget.AVG)
```

Comment:
If Max is more conservative should that be the default?

Reply:
Updated to use MAX as the default. Also added extra overhead which is pretty important for the memory to be able to grow once reduced.